mp4parse_capi/examples/test.cc

```diff
@@ -39,36 +39,34 @@ intptr_t io_read(uint8_t *buffer, uintptr_t size, void *userdata)
   return r;
 }
 
-void test_new_parser()
-{
-  int dummy_value = 42;
-  Mp4parseIo io = { abort_read, &dummy_value };
-  Mp4parseParser *parser = mp4parse_new(&io);
-  assert(parser != nullptr);
-  mp4parse_free(parser);
-  assert(dummy_value == 42);
-}
-
 void test_arg_validation()
 {
-  Mp4parseParser *parser = mp4parse_new(nullptr);
+  Mp4parseParser *parser = mp4parse_new(nullptr, nullptr);
+  assert(parser == nullptr);
+
+  Mp4parseStatus rv = MP4PARSE_STATUS_INVALID;
+  parser = mp4parse_new(nullptr, &rv);
+  assert(rv == MP4PARSE_STATUS_BAD_ARG);
   assert(parser == nullptr);
 
   Mp4parseIo io = { nullptr, nullptr };
-  parser = mp4parse_new(&io);
+  rv = MP4PARSE_STATUS_INVALID;
+  parser = mp4parse_new(&io, &rv);
+  assert(rv == MP4PARSE_STATUS_BAD_ARG);
   assert(parser == nullptr);
 
   io = { abort_read, nullptr };
-  parser = mp4parse_new(&io);
+  rv = MP4PARSE_STATUS_INVALID;
+  parser = mp4parse_new(&io, &rv);
```

Comment:
`abort_read` only existed to verify that initializing a parser didn't call into the IO context.  That's no longer true, so it's obsolete and can be removed now.